public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ArgQueryEditor/ArgQueryEditor.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import createMockDatasource from '../../__mocks__/datasource';
+import createMockQuery from '../../__mocks__/query';
+
+import ArgQueryEditor from './ArgQueryEditor';
+
+jest.mock('@grafana/runtime', () => ({
+  ...(jest.requireActual('@grafana/runtime') as unknown as object),
+  getTemplateSrv: () => ({
+    replace: (val: string) => {
+      return val;
+    },
+  }),
+}));
+
+const variableOptionGroup = {
+  label: 'Template variables',
+  options: [],
+};
+
+const defaultProps = {
+  query: createMockQuery(),
+  datasource: createMockDatasource(),
+  variableOptionGroup: variableOptionGroup,
+  onChange: jest.fn(),
+  setError: jest.fn(),
+};
+
+describe('ArgQueryEditor', () => {
+  it('should render', async () => {
+    render(<ArgQueryEditor {...defaultProps} />);
+    expect(await screen.findByTestId('azure-monitor-arg-query-editor-with-experimental-ui')).toBeInTheDocument();
+  });
+
+  it('should select a subscription from the fetched array', async () => {
+    const datasource = createMockDatasource({
+      getSubscriptions: jest.fn().mockResolvedValue([{ value: 'foo' }]),
+    });
+    const onChange = jest.fn();
+    render(<ArgQueryEditor {...defaultProps} datasource={datasource} onChange={onChange} />);
+    expect(await screen.findByTestId('azure-monitor-arg-query-editor-with-experimental-ui')).toBeInTheDocument();
+    expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ subscriptions: ['foo'] }));
+  });
+
+  it('should select a subscription from existing query', async () => {
+    const onChange = jest.fn();
+    const query = createMockQuery({
+      subscriptions: ['bar'],
+    });
+    render(<ArgQueryEditor {...defaultProps} onChange={onChange} query={query} />);
+    expect(await screen.findByTestId('azure-monitor-arg-query-editor-with-experimental-ui')).toBeInTheDocument();
+    expect(onChange).not.toHaveBeenCalledWith(expect.objectContaining({ subscriptions: ['foo'] }));
```

Comment:
Nit: This assertion is unneeded as there's no reference to `subscriptions: ['foo']` in this test.